apps/doop/vite.config.ts

```diff
@@ -0,0 +1,53 @@
+import { defineConfig } from "vite"
+import react from "@vitejs/plugin-react"
+import tailwindcss from "tailwindcss"
+import autoprefixer from "autoprefixer"
+
+export default defineConfig(({ mode }) => {
+  const sharedConfig = {
+    root: "./",
+
+    define: {
+      "process.env": {},
+    },
+
+    plugins: [react()],
+    css: {
+      postcss: {
+        plugins: [tailwindcss, autoprefixer],
+      },
+    },
+
+    server: {
+      host: "0.0.0.0",
+      port: parseInt(process.env.PORT || "3000"),
+    },
+  }
+
+  // with vite it is possible to have different configurations based on the mode
+  // we can use this to create a static build for previewing the app in github pages
+  // and also to create a docker image for the standalone app
+  if (mode === "static") {
+    return {
+      ...sharedConfig,
+      base: "./", // Relative Path in Generated index.html
```

Comment:
Could you clarify this line. I found it cool :) but would like to know more what exactly you try to achieve here.

Reply:
Sure! This is necessary to load the application (js part) relative to the index.html file. For example:

```
<script type="module" crossorigin src="./assets/index-otWdiz8X.js"></script>
```
instead of
```
<script type="module" crossorigin src="/assets/index-otWdiz8X.js"></script>
```

If the path is not relative, we run into issues when previewing the app, such as on GitHub Pages: https://cloudoperators.github.io/juno/pr-preview/pr-464/<application>